build.gradle

```diff
@@ -48,10 +50,13 @@ dependencies {
 
     compileOnly 'com.destroystokyo.paper:paper-api:1.16.4-R0.1-SNAPSHOT'
     compileOnly 'net.md-5:bungeecord-api:1.16-R0.4-SNAPSHOT'
+
     implementation 'org.jetbrains:annotations:20.1.0'
     implementation 'com.google.code.gson:gson:2.8.6'
     implementation 'io.github.classgraph:classgraph:4.8.92'
     implementation 'com.github.seancfoley:ipaddress:5.3.3'
+    implementation 'com.squareup.okhttp3:okhttp:4.9.0'
+    implementation 'commons-validator:commons-validator:1.7'
```

Comment:
Please use Guava's preconditions rather than shading in deps to StickyAPI.

Reply:
Please explain

Reply:
See the [JavaDocs for preconditions](https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html). This is included in both bukkit and bungee so will work at runtime. No need to shade in extra dependencies.

Reply:
Unless you're intending to use this for a completely different reason.

Reply:
Validator is used primarily to check url formatting for validity, and determine if a URL is validly formatted (and I think it also checks if it's resolvable). There are a couple other ways I plan to use it.

Reply:
We can also theoretically call the minimize() function to reduce jar size

Reply:
> Validator is used primarily to check url formatting for validity, and determine if a URL is validly formatted (and I think it also checks if it's resolvable). There are a couple other ways I plan to use it.

Google preconditions does thissssss - just include a URL regex.

Reply:
> We can also theoretically call the minimize() function to reduce jar size

This has side effects, we've already tried it.

Reply:
Do we even need to validate the URL ourselves? OkHttp should throw an exception when unable to parse a URL, and if we don't want to use OkHttp, a regular expression will do for URL validity

Reply:
We could also probably test URL validity by attempting to construct a URL object using the string and seeing if a MalformedURLException is thrown